Realm/Tests/QueryTests.m

```diff
@@ -180,8 +180,9 @@ - (void)testPredicateNotSupported
     // LinkList equality is unsupport since the semantics are unclear
     XCTAssertThrows(([ArrayOfAllTypesObject objectsWhere:@"ANY array = array"]));
 
-    // subquery
-    XCTAssertThrows(([ArrayOfAllTypesObject objectsWhere:@"SUBQUERY(array, $obj, $obj.intCol = 5).@count > 1"]));
+    // Unsupported variants of subqueries.
+    RLMAssertThrowsWithReasonMatching(([ArrayOfAllTypesObject objectsWhere:@"SUBQUERY(array, $obj, $obj.intCol = 5).@count == array.@count"]), @"SUBQUERY.*compared with a constant number");
+    RLMAssertThrowsWithReasonMatching(([ArrayOfAllTypesObject objectsWhere:@"SUBQUERY(array, $obj, $obj.intCol = 5) == 0"]), @"SUBQUERY.*immediately followed by .@count");
```

Comment:
AFAICT a common subquery pattern would be `prop IN SUBQUERY(...)` so it'd be nice to have a test to confirm we're throwing an exception with a helpful message in that case.